File: 392-is-subsequence/is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        left,right =0,0
        s_L= list(s)
        t_L= list(t)
        
        while left<len(t_L):
            if right<len(s_L) and s_L[right] == t_L[left]:
                right +=1
            left +=1
        return right == len(s_L)




        
        

File: 392-is-subsequence/is-subsequence.java
 class Solution {
    public boolean isSubsequence(String s, String t) {
        int lenA = s.length();
        int lenB = t.length();
        int slow = 0;
        int fast = 0;
        while(slow<lenA){  
             if(fast == lenB){
                return false;
            }
            char first = s.charAt(slow);
            char second = t.charAt(fast);
            if(first != second){
                fast++;
            }
            else{
                slow++;
                fast++;
            }

         
        }return true;
    }
}